src/SDKs/Azure.Core/data-plane/Azure.Core.Tests/UseSyncMethodsInterceptor.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Castle.DynamicProxy;
+
+namespace Azure.Core.Tests
+{
+    /// <summary>
+    /// This interceptor forwards the async call to a sync method call with the same arguments
+    /// </summary>
+    public class UseSyncMethodsInterceptor : IInterceptor
+    {
+        private const string AsyncSuffix = "Async";
+
+        private readonly MethodInfo TaskFromResultMethod = typeof(Task)
+            .GetMethod("FromResult", BindingFlags.Static | BindingFlags.Public);
+
+        private readonly MethodInfo TaskFromExceptionMethod = typeof(Task)
+            .GetMethods(BindingFlags.Static | BindingFlags.Public)
+            .Single(m => m.Name == "FromException" && m.IsGenericMethod);
+
+        public void Intercept(IInvocation invocation)
+        {
+            var methodName = invocation.Method.Name;
+            if (!methodName.EndsWith(AsyncSuffix))
+            {
+                throw new InvalidOperationException("Async method call expected");
+            }
+
+            var nonAsyncMethodName = methodName.Substring(0, methodName.Length - AsyncSuffix.Length);
+
+            var types = invocation.Method.GetParameters().Select(p => p.ParameterType).ToArray();
+
+            var methodInfo = invocation.TargetType.GetMethod(nonAsyncMethodName, BindingFlags.Public | BindingFlags.Instance, null, types, null);
```

Comment:
You need to filter out CancellationToken somewhere around here

Reply:
I think the guidance for now is to take CancellationToken into both.

Reply:
Really?  I guess that surprises me because people almost never use CancellationTokens for async methods so I have no idea why someone calling a sync method would do so, but alright.

Maybe consider adding a helpful suggestion to include a CancellationToken on the Sync overload in your exception here then?

Reply:
It would be handled by parameter mismatch exception.

Reply:
Yes, and I'm just suggesting adding a note to remember your CancellationToken in the message for that exception - since the 98% use case will be folks who forgot to add it to their sync overload and we might as well make it as easy as possible to figure out when it happens.

Reply:
Will do.